src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -234,7 +234,7 @@ def _parse_image_argument(cmd, namespace):
         return 'urn'
 
     # 3 - unmanaged vhd based images?
-    if urlparse(namespace.image).scheme:
+    if urlparse(namespace.image).scheme == "https" and namespace.image.endswith(".vhd"):
```

Comment:
@yugangw-msft is this overkill? Is it safer to just check that the scheme is https (or even http)?
Also, I am not sure we have any tests for vm creation using vhd blob uri.

Reply:
i think this check is good, but maybe can be simplified to just check `://` is in the string? This way blob name not necessarily has to be "*.vhd" which is a bit too strict

Reply:
I agree, I will make an update to this.